File: java/src/main/java/io/fivexer/sdk/model/TaskPriority.java
package io.fivexer.sdk.model;

/**
 * The result of repricing a task.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class TaskPriority {
    private String id;
    private double priority;

    public String getId() { return id; }
    public double getPriority() { return priority; }
}

File: php/src/Model/TaskPriority.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * The result of repricing a task.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class TaskPriority
{
    public function __construct(
        public readonly string $id,
        public readonly float $priority,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            priority: (float) ($data['priority'] ?? 0),
        );
    }
}
